structuredheadings/plugin.js

```diff
@@ -217,13 +217,20 @@
 
         onClick: function (value) {
           editor.applyStyle(elementStyles[ value ]);
-          setLevel(editor, editor.elementPath().block);
-          editor.execCommand("reapplyStyle");
+          var block = editor.elementPath().block;
+          var previousHeader = getPreviousHeader(editor, block);
           if (value === "p") {
+            CKEDITOR.plugins.structuredheadings.clearAll(editor, block);
```

Comment:
there was an actual bug found after adding the test, where p tags would retain the autonumbering classes as well